src/lib/builders/delaystate-builder.ts

```diff
@@ -26,9 +26,13 @@ import { validate } from '../utils';
  */
 function delaystateBuildingFn(data: Specification.Delaystate): () => Specification.Delaystate {
   return () => {
-    data.type = 'delay';
-    validate('Delaystate', data);
-    return data;
+    const result = {
+      type: 'delay',
+    } as Specification.Delaystate;
+
+    Object.assign(result, data);
+    validate('Delaystate', result);
+    return result;
```

Comment:
Since usedForCompensation is required now I don't think we have to set the value, I am thinking now that [setting the value `useForCompensation` in the constructor](https://github.com/serverlessworkflow/sdk-typescript/pull/114/files#diff-77b7743a1ea3c3b95c915b0c4c29d7014bea56839eb6113ba9065caa320ab20dR29) is not needed, I will have a look

Reply:
It might not be useful indeed, it should validate/be considerated false by default without having to specify it. @tsurdilo do you agree?